src/common/match.c

```diff
@@ -0,0 +1,19 @@
+// SPDX-License-Identifier: GPL-2.0-only
+#define _POSIX_C_SOURCE 200809L
+#include <glib.h>
+#include <stdbool.h>
+#include "common/match.h"
+
+bool
+match(const gchar *pattern, const gchar *string)
+{
+	GString *p = g_string_new(pattern);
+	GString *s = g_string_new(string);
+	g_string_ascii_down(p);
+	g_string_ascii_down(s);
+	bool ret = (bool)g_pattern_match_simple(p->str, s->str);
```

Comment:
Do we need the additional manual cast here?

Reply:
It appears not. Done.